code/compute_fc/compute_fc.py

```diff
@@ -0,0 +1,883 @@
+    )
+
+    logging.debug("Saving interpolated timeseries visual report at:")
+    logging.debug(f"\t{op.join(output, interpolate_saveloc)}")
+    os.makedirs(op.join(output, op.dirname(interpolate_saveloc)), exist_ok=True)
+    plt.savefig(op.join(output, interpolate_saveloc))
+
+
+def extract_and_denoise_timeseries(
+    func_filename,
+    atlas_filename,
+    verbose=2,
+    interpolate=False,
+    low_pass=None,
+    t_r=None,
+    output=None,
+    **kwargs,
+):
+    if not len(func_filename):
+        return [], []
+
+    logging.info(f"Extracting and denoising timeseries for {len(func_filename)} files.")
+    logging.debug(f"Denoising parameters are: {kwargs}")
+
+    # There is currently a bug in nilearn that prevent "load_confounds" from finding
+    # the confounds file if it contains any other BIDS entity than "ses" and "run".
+    # It should be fixed in release 0.13.
+    try:
+        confounds, sample_mask = load_confounds(
+            func_filename,
+            demean=False,
+            strategy=DENOISING_STRATEGY,
+            motion="basic",
+            **kwargs,
+        )
+    except ValueError:
+        logging.warning(
+            "Nilearn could not find the confounds file (this is likely due to a"
+            " bug in nilearn.interface.fmriprep.load_confouds that should be fixed in"
+            " release 0.13, see nilearn issue #3792)."
+        )
+        logging.warning("Searching manually ...")
+
+        confounds, sample_mask = get_confounds_manually(
+            func_filename,
+            demean=False,
+            strategy=DENOISING_STRATEGY,
+            motion="basic",
+            **kwargs,
+        )
+
+    # The outputs of "load_confounds" will not be in a list if
+    # "func_filename" is a list with one element.
+    if not isinstance(confounds, list):
+        confounds = [confounds]
+    if not isinstance(sample_mask, list):
+        sample_mask = [sample_mask]
+
+    if interpolate:
+        interpolate_and_denoise_timeseries(
+            func_filename,
+            atlas_filename,
+            confounds,
+            sample_mask,
+            t_r=t_r,
+            low_pass=low_pass,
+            output=output,
+            verbose=verbose,
+        )
+
+    time_series = fit_transform_patched(
+        func_filename,
+        atlas_filename,
+        confounds,
+        sample_mask,
+        low_pass=low_pass,
+        t_r=t_r,
+        standardize="zscore_sample",
+        verbose=verbose,
+        reports=True,
+        n_jobs=8,
+    )
+
+    return time_series, confounds
+
+
+def get_fc_strategy(strategy="sparse inverse covariance"):
+    connectivity_kind = "correlation"
+    connectivity_label = "correlation"
+    estimator = LedoitWolf(store_precision=False)
+
+    if strategy not in ["cor", "corr", "correlation"]:
+        connectivity_kind = "precision"
+        connectivity_label = "sparseinversecovariance"
+        estimator = GraphicalLassoCV(alphas=6, max_iter=1000)
```

Comment:
I find it a bit confusing to base the conditions on `not in`. What about?

```suggestion
    if strategy in ["cor", "corr", "correlation"]:
        connectivity_kind = "correlation"
        connectivity_label = "correlation"
        estimator = LedoitWolf(store_precision=False)
    elif strategy in ["sparse", "sparse inverse covariance"]:
        connectivity_kind = "precision"
        connectivity_label = "sparseinversecovariance"
        estimator = GraphicalLassoCV(alphas=6, max_iter=1000)
```
To be completed to include the last condition

Reply:
I believe this will not work if you want to use the `covariance` method as it will require `strategy` to match the `elif` statement. I'll fix this in the next commit.